### PR TITLE
Support sensors

### PR DESCRIPTION
**Pull request by hobinjk**
Adds support for Hue motion, temperature, light, and daylight sensors

### CODE REVIEW

File: philips-hue-adapter.js [hunk 1]
```diff
@@ -103,35 +111,55 @@ class PhilipsHueDevice extends Device {
...
+    this.name = device.name;
+
+    if (deviceId.startsWith('sensors')) {
+      if (device.state.hasOwnProperty('presence')) {
+        this.type = Constants.THING_TYPE_BINARY_SENSOR;
+        this.properties.set('on',
+          new PhilipsHueProperty(this, 'on', {type: 'boolean'},
+                                 device.state.presence));
+      } else if (device.state.hasOwnProperty('temperature')) {
+        this.type = Constants.THING_TYPE_UNKNOWN_THING;
```
Comment (mrstegeman) [inline philips-hue-adapter.js]: Maybe this would be more appropriate as `THING_TYPE_MULTI_LEVEL_SENSOR`.

File: philips-hue-adapter.js [hunk 2]
```diff
@@ -316,61 +362,91 @@ class PhilipsHueAdapter extends Adapter {
...
+      device.update(deviceState);
+      return;
+    }
+
+    if (deviceId.startsWith('sensors')) {
+      if (!SUPPORTED_SENSOR_TYPES[deviceState.type]) {
+        return;
+      }
+    }
+    const normalizedId = deviceId.replace(/lights\//g, '')
```
Comment (mrstegeman) [inline philips-hue-adapter.js]: What about `sensors/`?

Comment (hobinjk) [reply philips-hue-adapter.js]: This replacement is to make existing lights' ids not change. Without this replacement the light formerly known as philips-hue-6893894384-1 would be renamed to philips-hue-6893894384-lights-1 which would require anyone who is currently using the adapter to remove the old light and re-add it to update the id.

Comment (mrstegeman) [reply philips-hue-adapter.js]: Ah, fair enough.

File: philips-hue-adapter.js (hunk 1 above)
Comment (hobinjk) [reply philips-hue-adapter.js]: multiLevelSensor is specified as taking values from 0-100 but the temperature may fall out of that range, especially when it's this time of the year where I live.

Comment (mrstegeman) [reply philips-hue-adapter.js]: Right, I forgot we went with percentage instead of a bare number.

**Comment by hobinjk** [comment]
Thanks for the review!

**Pull request merged**